transport/src/main/java/io/netty5/channel/DefaultChannelHandlerContext.java

```diff
@@ -820,13 +822,14 @@ void callHandlerRemoved() throws Exception {
             }
         } finally {
             // Mark the handler as removed in any case.
+            removed = true;
             handlerState = REMOVE_COMPLETE;
```

Comment:
I would flip these two around so the `handlerState` update can piggy-back on the volatile store to `removed`, even though it's not strictly necessary.
With a VarHandle we could also make just these two `handlerState` loads and stores volatile.